packages/common-components/src/Toasts/ToastContext.tsx

```diff
@@ -0,0 +1,209 @@
+  toasts: Toast[]
+}
+
+const ToastContext = React.createContext<ToastContext | undefined>(
+  undefined
+)
+
+const ToastProvider = ({
+  children,
+  autoDismiss = true,
+  defaultPosition = "top-right",
+  dismissTimeout = 5000
+}: ToastContextProps) => {
+  const classes = useStyles()
+  const [toasts, setToasts] = useState<Toast[]>([])
+
+  const removeToast = useCallback((toastId: string) => {
+    setToasts(toasts.filter((toast) => toast.id !== toastId))
+  }, [toasts])
+
+  const addToast = useCallback((toastParams: ToastParams) => {
+    const id = uuidv4()
+    setToasts((toasts) => ([
+      ...toasts,
+      { id,
+        ...toastParams,
+        toastPosition: toastParams.toastPosition || defaultPosition
+      }
+    ]))
+
+    const isProgressToast = toastParams.progress !== undefined
+    const shouldDismiss = toastParams.autoDismiss !== undefined ? toastParams.autoDismiss : autoDismiss
+    const dismissTimeOut = toastParams.dismissTimeout || dismissTimeout
+
+    if (shouldDismiss && !isProgressToast) {
+      setTimeout(() => {
+        setToasts((toasts) => toasts.filter((toast) => toast.id !== id))
+      }, dismissTimeOut)
+    }
+
+    return id
+  }, [autoDismiss, dismissTimeout, defaultPosition])
+
+  const updateToast = useCallback((toastId: string, toastParams: ToastParams, startDismissal?: boolean) => {
+    const dismissTimeOut = toastParams.dismissTimeout || dismissTimeout
+    if (startDismissal) {
+      setTimeout(() => {
+        setToasts((toasts) => toasts.filter((toast) => toast.id !== toastId))
+      }, dismissTimeOut)
+    }
+    setToasts((toasts) => toasts.map((toast) => toast.id === toastId ? { ...toast, ...toastParams } : toast))
+  }, [dismissTimeout])
+
+  const positionWiseToasts = useMemo(() => ([
+    {
+      toasts: toasts.filter((toast) => toast.toastPosition === "top-right"),
+      identifier: "topRight"
+    },
+    {
+      toasts: toasts.filter((toast) => toast.toastPosition === "top-left"),
+      identifier: "topLeft"
+    },
+    {
+      toasts: toasts.filter((toast) => toast.toastPosition === "bottom-right"),
+      identifier: "bottomRight"
+    },
+    {
+      toasts: toasts.filter((toast) => toast.toastPosition === "bottom-left"),
+      identifier: "bottomLeft"
+    }
+  ]), [toasts])
+
+  return (
+    <ToastContext.Provider
+      value={{
+        addToast,
+        updateToast,
+        removeToast,
+        toasts
+      }}
+    >
+      {positionWiseToasts.map((positionedToasts) => (
+        !!positionedToasts.toasts.length && (
+          <div
+            key={positionedToasts.identifier}
+            className={clsx(
+              classes.toastWrapper,
+              classes[`${positionedToasts.identifier}Container`]
+            )}
+          >
+            {positionedToasts.toasts.map((toast) => (
+              <div
+                key={toast.id}
+                className={classes[`${positionedToasts.identifier}AppearBox`]}
+              >
```

Comment:
Updates types, variable names
  
  ```suggestion
const positionedToasts: Record<ToastPosition, Array<Toast>> = useMemo(() => ({
    topRight: toasts.filter((toast) => toast.toastPosition === "topRight"),
    topLeft: toasts.filter((toast) => toast.toastPosition === "topLeft"),
    bottomRight: toasts.filter((toast) => toast.toastPosition === "bottomRight"),
    bottomLeft: toasts.filter((toast) => toast.toastPosition === "bottomLeft"),
  }), [toasts])
  return (
    <ToastContext.Provider
      value={{
        addToast,
        updateToast,
        removeToast,
        toasts
      }}
    >
      {(Object.keys(positionedToasts) as ToastPosition[]).map((position) => (
        !!positionedToasts[position].length && (
          <div
            key={position}
            className={clsx(
              classes.toastWrapper,
              classes[`${position}Container`]
            )}
          >
            {positionedToasts[position].map((toast) => (
              <div
                key={toast.id}
                className={classes[`${position}AppearBox`]}
              >
```